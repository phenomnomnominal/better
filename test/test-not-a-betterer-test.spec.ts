import { createFixture } from './fixture';

describe('betterer', () => {
  it(`should throw if it doesn't return a BettererTest`, async () => {
    const { betterer } = await import('@betterer/betterer');

    const { paths, logs, cleanup } = await createFixture('test-not-a-betterertest', {
      '.betterer.js': `

module.exports = {
  test: () => {}
};
      `
    });

    const configPaths = [paths.config];
    const resultsPath = paths.results;

    await expect(async () => await betterer({ configPaths, resultsPath, workers: false })).rejects.toThrow();

    expect(logs).toMatchSnapshot();

    await cleanup();
  });
});
