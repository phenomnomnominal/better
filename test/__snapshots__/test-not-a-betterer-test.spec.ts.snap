// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betterer should throw if it doesn't return a BettererTest 1`] = `
Array [
  "
Error: \\"test\\" must return a \`BettererTest\`.
",
]
`;
